build.gradle

```diff
@@ -1,30 +1,67 @@
 apply plugin: "cpp"
 
+def boost_home = System.getenv("BOOST_HOME")
 model {
-    binaries {
-      all {
-          linker.args "-lpthread"
+  binaries {
+    all {
+      if (buildType == buildTypes.debug) {
+        cppCompiler.args "-g"
       }
+
+      cppCompiler.args "-Wextra"
+      cppCompiler.args "-Wall"
```

Comment:
@AGaliuzov, `Wextra` = `Wall` + some additional checks. Thus `Wall` is redundant.